mmv1/third_party/terraform/services/vmwareengine/resource_vmwareengine_subnet_test.go

```diff
@@ -0,0 +1,89 @@
+package vmwareengine_test
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+)
+
+func TestAccVmwareengineSubnet_vmwareEngineUserDefinedSubnetUpdate(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"region":        "southamerica-west1", // using region with low node utilization.
+		"random_suffix": acctest.RandString(t, 10),
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testVmwareEngineSubnetConfig(context, "192.168.1.0/26"),
+				Check: resource.ComposeTestCheckFunc(
+					acctest.CheckDataSourceStateMatchesResourceStateWithIgnores("data.google_vmwareengine_subnet.ds", "google_vmwareengine_subnet.vmw-engine-subnet", map[string]struct{}{}),
+				),
+			},
+			{
+				ResourceName:            "google_vmwareengine_subnet.vmw-engine-subnet",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"parent", "name"},
+			},
+			{
+				Config: testVmwareEngineSubnetConfig(context, "192.168.2.0/26"),
+			},
+			{
+				ResourceName:            "google_vmwareengine_subnet.vmw-engine-subnet",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"parent", "name"},
+			},
+		},
+	})
+}
+
+func testVmwareEngineSubnetConfig(context map[string]interface{}, ipCidrRange string) string {
+	context["ip_cidr_range"] = ipCidrRange
+	return acctest.Nprintf(`
+resource "google_vmwareengine_network" "subnet-nw" {
+  name        = "tf-test-subnet-nw%{random_suffix}"
+  location    = "global"
+  type        = "STANDARD"
+  description = "PC network description."
+}
+
+resource "google_vmwareengine_private_cloud" "subnet-pc" {
+  location    = "%{region}-a"
+  name        = "tf-test-subnet-pc%{random_suffix}"
+  description = "Sample test PC."
+  network_config {
+    management_cidr       = "192.168.0.0/24"
+    vmware_engine_network = google_vmwareengine_network.subnet-nw.id
+  }
+
+  management_cluster {
+    cluster_id = "tf-test-mgmt-cluster%{random_suffix}"
+    node_type_configs {
+      node_type_id = "standard-72"
+      node_count   = 3
+    }
+  }
+}
+
+resource "google_vmwareengine_subnet" "vmw-engine-subnet" {
+	name = "service-2"
+	parent =  google_vmwareengine_private_cloud.subnet-pc.id
+	ip_cidr_range = "%{ip_cidr_range}"
+}
+
+data "google_vmwareengine_subnet" ds {
+	name = "service-2"
+	parent = google_vmwareengine_private_cloud.subnet-pc.id
+	depends_on = [
+    google_vmwareengine_subnet.vmw-engine-subnet,
+  ]
+}
```

Comment:
The indentation should be 2 spaces.

The name of `google_vmwareengine_subnet` should follow the format of `tf-test-`

Reply:
`google_vmwareengine_subnet` are automatically created during the parent PC creation and follow the naming convention of `service-n` for user defined subnets and 'vsan', 'vmotion', etc for management subnets. They can only be modified through their ip_cidr_range.

Reply:
Does it depend on the configuration of the private cloud resource that if the subnet is the user defined or management subnet?

Reply:
How does the user know what name should be put into the configuration of the resource or data source?

Reply:
Each private cloud has a certain number of user defined and management subnets. These are predefined. ('service-1' to 'service-5' are the names for user defined subnets, while management subnets are named 'vsan', 'vmotion' 'system-management' etc. 

Management subnets are read only and are to be read via the data source, and are identified using parent and name. 

User defined subnets can be used through resources as well to update their 'ip_cidr_field'.

Reply:
These 12 subnets are automatically created with each PC creation.

Reply:
Thanks for the information. Can you please also add the information to `description`, `name` of the resource yaml file? And add more details to the `name` field in the file  mmv1/third_party/terraform/website/docs/d/vmwareengine_subnet.html.markdown?

Reply:
Done.